test/mocha/unhandled_rejections.js

```diff
@@ -677,6 +677,41 @@ if (windowDomEventSupported) {
 
         })
     });
+
+    if (typeof Worker !== "undefined") {
+        describe("dom events in a worker", function() {
+            var worker;
+            beforeEach(function () {
+                worker = new Worker("./worker.js");
+            });
+
+            afterEach(function () {
+                worker.terminate();
+            });
+
+            specify("are fired", function() {
+                return new Promise(function(resolve, reject) {
+                    var order = [];
+                    worker.onmessage = function (message) {
+                        switch(message.data) {
+                        case "unhandledrejection":
+                            order.push(1);
+                            break;
+                        case "rejectionhandled":
+                            order.push(2);
+                            resolve();
```

Comment:
didn't you mean to validate the order here? Also note that you might want to wrap the onmessage in a try/catch in case something throws there - reject.